pkg/confidence/utils.go

```diff
@@ -308,7 +316,7 @@ func ToObjectResolutionDetail(res InterfaceResolutionDetail, defaultValue interf
 			}
 		}
 
-		err := NewTypeMismatchResolutionError("Unable to convert response property to float")
+		err := NewTypeMismatchResolutionError("Unable to convert response property to object")
```

Comment:
@fabriziodemaria also here, but in generally about this part, I did not understand the true nature of ToObjectResolutionDetail. I'm probably missing some internal decision thinking, including about the package provider.

Reply:
The idea of `ToObjectResolutionDetail` is to be able to return the entire flag, rather than accessing a specific property of that flag. I think we still miss a strong specification regarding this API:

- What should happen when some/all the properties of the flag gotten from the backend are `null`? Should we replace only those null properties with the corresponding property values from the the defaultValue passed in?
- What should happen if the schema of the flag from the backend and the schema of the defaultValue passed in are different?

This problem is separate from the scope of the PR, so I suggest we go ahead and start merging these improvements!